### PR TITLE
Imitate stable's slider followcircle fade + scale animations

### PR DESCRIPTION
**Pull request by goodtrailer**
Continuation of #14832 - part 3

This is the meat of the original PR and addresses the differences between stable and lazer's followcircle fade/scale animations. First, stable's fade out + shrink animations are significantly slower at the end of sliders. Second, stable's fade in + expand animations are sped up on extremely short sliders, such that they are never interrupted by the slider ending. Such interruption results in even shorter fade out + shrink animations, since the followcircle disappears from a semi-appeared state (as opposed to disappearing from 100% visible like in stable).

Demonstration video: https://www.youtube.com/watch?v=eGvTqti5VlQ

### Summary
* Removes followcircle animations from `DrawableSlider` and `DrawableSliderBall`, since there they affect both default and legacy skins
* Copies above animations to `DefaultFollowCircle` and `DefaultSliderBall`, so the default skin is unaffected by this PR
* Adds new animations to `LegacyFollowCircle` that are slower at slider ends and aren't interrupted on short sliders

### Notes
Fade out is still a bit faster than stable (and the curve is noticeably steeper). This is because lazer sliders themselves fade out faster than stable, and followcircles are children to sliders. This requires changes to slider animations that deserve a separate PR (and IMO are a bit overkill, but maybe that's just me).

### CODE REVIEW

**Review by peppy — COMMENTED** [review]
At the point we are fiddling around with millisecond changes, it should probably match the stable implementation 1:1 (if that's what people need from it).

So rather than video comparisons we'd be checking against the stable implementation and matching line by line. If you want a copy of the relevant stable implementation feel free to ask, else we will need to do that in the review step.

File: osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
 
 namespace osu.Game.Rulesets.Osu.Skinning.Legacy
 {
     public class LegacyFollowCircle : CompositeDrawable
```
[the remark this answers is not in the document]
Comment (goodtrailer) [reply osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs]: Just pushed some changes which move the shared logic to an abstract base class and uses template methods for animations. I think that's better? I do feel a bit strange about the placement/location of the class, not sure if there's a better place where it belongs. (The animations are of course not final, I just wanted to know first if this is a better approach structurally before addressing your comment above)

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs]: Much better now.

**Comment by peppy** [comment]
Note that in stable, when tracking is interrupted by user error, the follow circle scales *outwards* not *inwards*. I think this looks nice, but it's not implemented in lazer yet (should probably be in both implementations).

Also, it only animates this tracking break at the next slider tick.

**Review by peppy — DISMISSED** [review]

**Review by peppy — APPROVED** [review]

File: osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs [hunk 2]
```diff
@@ -18,5 +20,36 @@ public LegacyFollowCircle(Drawable animationContent)
...
         }
+
+        protected override void OnTrackingChanged(ValueChangedEvent<bool> tracking)
+        {
+            Debug.Assert(ParentObject != null);
+
+            if (ParentObject.Judged)
+                return;
+
+            double remainingTime = ParentObject.HitStateUpdateTime - Time.Current;
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs:31]: https://github.com/ppy/osu/pull/19132

**Pull request merged**